node/src/components/consensus/protocols/highway.rs

```diff
@@ -471,76 +472,56 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
             .map_or(false, is_switch)
     }
 
-    // Check if we've made any progress since joining.
-    // If we haven't, we might have been left alone in the era and we should request the state from
-    // peers.
-    fn handle_progress_alert_timer(&mut self, now: Timestamp) -> ProtocolOutcomes<I, C> {
+    /// Request the latest state from a random peer.
+    fn handle_request_state_timer(&mut self, now: Timestamp) -> ProtocolOutcomes<I, C> {
         if self.evidence_only || self.finalized_switch_block() {
             return vec![]; // Era has ended. No further progress is expected.
         }
-        if self.last_panorama == *self.highway.state().panorama() {
-            info!(
-                instance_id = ?self.highway.instance_id(),
-                "no progress in the last {}, creating latest state request",
-                self.standstill_timeout,
-            );
-            // We haven't made any progress. Request latest panorama from peers and schedule
-            // standstill alert. If we still won't progress by the time
-            // `TIMER_ID_STANDSTILL_ALERT` is handled, it means we're stuck.
-            let mut outcomes = self.latest_state_request();
-            if self.shutdown_on_standstill {
-                outcomes.push(ProtocolOutcome::ScheduleTimer(
-                    now + self.standstill_timeout,
-                    TIMER_ID_STANDSTILL_ALERT,
-                ));
-            }
-            return outcomes;
-        }
-
-        if !self.shutdown_on_standstill {
-            debug!(
-                instance_id = ?self.highway.instance_id(),
-                "progress detected; not requesting latest state",
-            );
-            return vec![];
-        }
         debug!(
             instance_id = ?self.highway.instance_id(),
-            "progress detected; scheduling next standstill check in {}",
-            self.standstill_timeout,
+            "requesting latest state from random peer",
         );
-        // Record the current panorama and schedule the next standstill check.
-        self.last_panorama = self.highway.state().panorama().clone();
-        vec![ProtocolOutcome::ScheduleTimer(
-            now + self.standstill_timeout,
-            TIMER_ID_STANDSTILL_ALERT,
-        )]
+        // We haven't made any progress. Request latest panorama from peers and schedule
```

Comment:
Aren't we requesting the latest state regardless of whether we made any progress or not?